test/partial_shape.cpp

```diff
@@ -200,3 +200,66 @@ TEST(partial_shape, shapes_equal_both_complete_same_rank_different_dims)
 
     ASSERT_FALSE(ps1.compatible(ps2));
 }
+
+TEST(partial_shape, from_shape)
+{
+    Shape s{2, 4, 6, 8};
+    PartialShape ps1{s};
+
+    // TODO(amprocte): No way to examine contents of ps1 yet.
```

Comment:
Regarding this TODO: the next PR will be adding some things like `operator[]` on `PartialShape`, etc. I can update these tests in that PR.